Source/JavaScriptCore/runtime/CachedTypes.cpp

```diff
@@ -304,8 +304,9 @@ size_t Decoder::size() const
 ptrdiff_t Decoder::offsetOf(const void* ptr)
 {
     const uint8_t* addr = static_cast<const uint8_t*>(ptr);
-    ASSERT(addr >= m_cachedBytecode->data() && addr < m_cachedBytecode->data() + m_cachedBytecode->size());
-    return addr - m_cachedBytecode->data();
+    auto cachedBytecodeSpan = m_cachedBytecode->span();
+    ASSERT(addr >= cachedBytecodeSpan.data() && addr < (cachedBytecodeSpan.data() + cachedBytecodeSpan.size()));
```

Comment:
You'd think there would be an idiomatic way to write this. Or should we add a helper function if the standard library doesn't have it?

Reply:
Maybe I should add something like this in StdLibExtras.h:
```
bool isInSpanRange(std::span<T> span, T* address);
```

Reply:
If there isn't something like that in the standard library, I think it would be good. I'd want it to work even when the const-ness of the pointer and the span differed. Maybe the ranges part of the standard library has this.

Reply:
I think it's ambiguous whether a point is in a range of it's at the end. That makes this function name annoying.

Reply:
Ok, if it's confusing then I'm going back to not having this function. It is only used in one place at the moment anyway.